CMakeLists.txt

```diff
@@ -7,6 +7,10 @@ project(box2d VERSION 2.4.0)
 
 set_property(GLOBAL PROPERTY USE_FOLDERS ON)
 
+set(CMAKE_CXX_VISIBILITY_PRESET hidden)
```

Comment:
You should probably add `set(CMAKE_VISIBILITY_INLINES_HIDDEN 1)` as well.

Reply:
Many functions are defined inline in the headers, I do not think it's a good idea to make them hidden by default. In fact, all inline functions are public, there is no private inline.

Reply:
Yes, that's exactly why you'd want to enable this option. Otherwise the inline functions might get exported. Those aren't needed since they are inline (i.e. the code will be available to consuming projects).